### DISCUSSION TITLE
Detecting what triggered "paths" directive

### DISCUSSION THREAD

**Discussion by mfridman** [community/community#26742] [Actions]
<p>Is it possible to detect <em>what</em> triggered a workflow when specifying paths?</p>
<p>e.g., say we have this directory structure</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - master
    paths:
      - "cluster_1/namespace_1/**"
      - "cluster_1/namespace_2/**"
      - "cluster_2/namespace_1/**"
      - "cluster_3/namespace_1/**"
</code></pre>
<p>And a merge was made to master with changes to <code>cluster_2/namespace_1/**</code> … is it possible to get this info from any of the GITHUB env variables, or context?</p>
<p>I know I can grab the merged PR and reach out to the API to get the necessary info, just wondering if there is a simpler way?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mfridman">@mfridman</a>,</p>
<p>There is no predefined, special variables to get the information you require.</p>
<p>For your case, generally you can do with the steps below:</p>
<ol>
<li>
<p>List all the changed files contained in current push. You can use the “<a href="https://git-scm.com/docs/git-diff" rel="noopener nofollow ugc"><strong>git diff</strong></a>” command or from the “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-a-commit" rel="noopener nofollow ugc"><strong>Get a commit</strong></a>” endpoint to list all the changed files.</p>
</li>
<li>
<p>Use the patterns of the path filters you set for the push event in the workflow file to compare with the path of each changed file in the list.</p>
</li>
<li>
<p>The files that can match the patterns are the files which triggered the push event.</p>
</li>
</ol>

**Comment by mcassaniti** [comment]
<p>How would I go about requesting this as a feature of GitHub workflows? I have several duplicate workflows that only differ based on the triggered path.</p>
<pre><code class="lang-yaml">---
  name: Test influxdb
  on:
    pull_request:
      paths:
        - influxdb/**

  jobs:
    plan:
      name: Test influxdb
      runs-on: ubuntu-latest

      steps:
        - name: Clone repository
          uses: actions/checkout@v2

        - name: Configure AWS Credentials
          uses: aws-actions/configure-aws-credentials@v1
          with:
            aws-access-key-id: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY }}
            aws-secret-access-key: ${{ secrets.TERRAFORM_AWS_SECRET_KEY }}
            aws-region: ap-southeast-2

        - name: Plan Terraform
          uses: dflook/terraform-plan@v1
          with:
            # The below variable is the only thing to change
            path: influxdb
            workspace: prod
            add_github_comment: false
</code></pre>
